VitalHealthConnect/src/main/java/io/tryvital/vitalhealthconnect/VitalHealthConnectManager.kt

```diff
@@ -477,18 +486,22 @@ class VitalHealthConnectManager private constructor(
             ).build()
         )
 
-        observerJob?.cancel()
-        observerJob = CoroutineScope(Dispatchers.Main).launch {
-            launch {
-                work.workInfosLiveData.observeForever { workInfos ->
-                    updateStatusFromJob(workInfos)
-                }
+        taskScope.launch(Dispatchers.Main) {
+            val observer = androidx.lifecycle.Observer<List<WorkInfo>> { workInfos ->
+                updateStatusFromJob(workInfos)
             }
+            work.workInfosLiveData.observeForever(observer)
             work.enqueue()
+
+            try {
+                awaitCancellation()
+            } catch (e: Throwable) {
+                work.workInfosLiveData.removeObserver(observer)
+            }
```

Comment:
Should we do something with `e`?

Reply:
not necessary because we expect only `CancellationException` being thrown by `awaitCancellation()`, which indicates the scope @ manager level has been cancelled.